pkg/controllers/podfacts.go

```diff
@@ -205,30 +205,48 @@ func (p *PodFacts) collectPodByStsIndex(ctx context.Context, vdb *vapi.VerticaDB
 
 // checkIsInstalled will check a single pod to see if the installation has happened.
 func (p *PodFacts) checkIsInstalled(ctx context.Context, vdb *vapi.VerticaDB, pf *PodFact) error {
-	if pf.isPodRunning {
-		fn := paths.GenInstallerIndicatorFileName(vdb)
-		if stdout, stderr, err := p.PRunner.ExecInPod(ctx, pf.name, names.ServerContainer, "cat", fn); err != nil {
-			if !strings.Contains(stderr, "cat: "+fn+": No such file or directory") {
-				return err
-			}
+	if !pf.isPodRunning {
+		pf.isInstalled = tristate.None
+		return nil
+	}
+
+	// If initPolicy is ScheduleOnly, there is no install indicator since the
+	// operator didn't initiate it.  We are going to do based on the existence
+	// of admintools.conf.
+	if vdb.Spec.InitPolicy == vapi.CommunalInitPolicyScheduleOnly {
+		if _, _, err := p.PRunner.ExecInPod(ctx, pf.name, names.ServerContainer, "test", "-f", paths.AdminToolsConf); err != nil {
 			pf.isInstalled = tristate.False
+		} else {
+			pf.isInstalled = tristate.True
+		}
 
-			// Check if there is a stale admintools.conf
-			cmd := []string{"ls", paths.AdminToolsConf}
-			if _, stderr, err := p.PRunner.ExecInPod(ctx, pf.name, names.ServerContainer, cmd...); err != nil {
-				if !strings.Contains(stderr, "No such file or directory") {
-					return err
-				}
-				pf.hasStaleAdmintoolsConf = false
-			} else {
-				pf.hasStaleAdmintoolsConf = true
+		// We can't reliably set compat21NodeName because the operator didn't
+		// originate the install.  We will intentionally leave that blank.
+		pf.compat21NodeName = ""
```

Comment:
No change required, just trying to confirm my understanding of the functionalities: does this mean re_ip will not be automated on the schedule only pods? I recall compat21NodeName is used for re_ip when db is down but would like to double check if my memory is correct.

Reply:
nvm, I saw the changes in restart_reconcile and I think my question is answered.